chain/pool/txpool.go

```diff
@@ -30,6 +30,20 @@ func NewTxPool() *TxnPool {
 }
 
 func (tp *TxnPool) AppendTxnPool(txn *transaction.Transaction) error {
+	sender, err := txn.GetProgramHashes()
+	if err != nil {
+		return err
+	}
+
+	list, err := tp.getOrNewList(sender[0])
+	if err != nil {
+		return err
+	}
+
+	if _, err := list.Get(txn.UnsignedTx.Nonce); err != nil && list.Full() {
```

Comment:
Should this be err == nil ?

Reply:
that is : if the same nonce txn does not existed in txlist and txnlist is full, reject this txn.

Reply:
You are right